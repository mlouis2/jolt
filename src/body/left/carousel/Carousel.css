.Carousel {
  width: 100%;
}

.Carousel img {
  background-color: black;
  float: left;
  width: 100px;
  height: 100px;
  border-bottom: 5px solid black;
  border-right: 5px solid black;
}

.Carousel button {
  visibility: hidden;
}

.Carousel li {
    list-style-type: none;
}

.Carousel ul {
  padding: 0px;
}
