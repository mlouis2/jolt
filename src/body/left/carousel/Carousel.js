import React, {useState, useEffect, useRef} from 'react'
import { Carousel } from 'react-responsive-carousel';
import { Pokemon } from './Pokemon'
import DetailedView from '../../right/DetailedView'

import './Carousel.css'
import './SearchBar.css'

const arrowKeyCodes = [37, 38, 39, 40];

function ResponsiveCarousel(props) {
    const pokemonList = props.pokemonList;
    const [filteredList, setFilteredList] = useState([]);
    const [search, setSearch] = useState({input: ''});
    const [currentPokemonIndex, setCurrentPokemonIndex] = useState(0);

    const indexRef = useRef();

    function onFocusChange(e) {
        indexRef.current = e;
        setCurrentPokemonIndex(e);
    }

    function onClick(e) {
        if (indexRef.current !== e) {
            indexRef.current = e;
            setCurrentPokemonIndex(e);
        }
    }

    useEffect(() => {
        setFilteredList(pokemonList);
    }, [pokemonList])

    return (
        <div>
        <div className="SearchContainer">
        <SearchInput />
        </div>
        <Carousel
        className="Carousel"
        axis="vertical"
        onChange={onFocusChange}
        onClickItem={onClick}
        selectedItem={indexRef.current}
        showArrows={false}
        showThumbs={false}
        showIndicators={false}
        showStatus={false}
        useKeyboardArrows={true}
        width="100%"
        centerMode={true}>
        {(search.input !== '' ? filteredList : pokemonList).map(p => {
            return (
                <div key={p.index}>
                {Pokemon(p.sprite, p.name, p.number, p.types, p.description, p.index)}
                </div>
            )
        })}
        </Carousel>
        <DetailedView currentPokemon={filteredList[currentPokemonIndex]} pokemonList={pokemonList} />
        </div>
    );

    function SearchInput() {

        const searchRef = useRef();

        useEffect(() => {
            window.addEventListener("keydown", onKeyPressed);
            const searchBar = document.getElementById("searchBar");
            if (searchBar) {
                searchBar.focus();
            }
        }, [])

        function onKeyPressed(event) {
            if (arrowKeyCodes.includes(event.keyCode)){
                document.getElementById("searchBar").blur();
            }
        }

        function handleTextInput(e) {
            const prev = searchRef.current;
            searchRef.current = e.target.value;
            setSearch({input: e.target.value});
            if (e.target.value === "") {
                if (prev !== '') {
                    setFilteredList(pokemonList)
                }
            } else {
                handleSearch();
            }
        }

        function handleSearch() {
            setFilteredList(pokemonList.filter(pokemon => pokemon.name.toLowerCase().includes(searchRef.current.toLowerCase())));
            indexRef.current = 0;
        }

        return (
            <div className="SearchBarContainer">
            <input type="text" className="SearchBar" id="searchBar" onChange={handleTextInput} placeholder="Search for a PokÃ©mon!" value={search.input}>
            </input>
            </div>
        )
    }
}

export {
    ResponsiveCarousel
}
