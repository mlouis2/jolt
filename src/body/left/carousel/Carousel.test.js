import React from 'react'
import ReactDOM from 'react-dom'
import Carousel from './Carousel'
import { render, waitForElement, fireEvent } from "@testing-library/react";

it('renders without crashing', () => {
     const carousel = render(<Carousel />)
     expect(carousel).toMatchSnapshot()
})

it('allows for user to search', async () => {
     const { getByTestId } = render(<Carousel />)
     const searchBar = await waitForElement(() =>
       getByTestId('searchBar'),
     )
     fireEvent.click(searchBar)
     fireEvent.keyDown(searchBar, { key: 'A', code: 65, charCode: 65 })
})
