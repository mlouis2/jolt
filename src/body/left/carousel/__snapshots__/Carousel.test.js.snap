// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows for user to type in search bar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="searchContainer"
      >
        <div
          class="searchBarContainer"
        >
          <input
            class="searchBar"
            data-testid="searchBar"
            id="searchBar"
            placeholder="Search by name, type, or number!"
            type="text"
            value="A"
          />
        </div>
        <div
          class="carousel"
        >
          <div
            class="carousel carousel-slider"
            style="width: 100%;"
          >
            <button
              class="control-arrow control-prev control-disabled"
              type="button"
            />
            <div
              class="slider-wrapper axis-vertical"
            >
              <ul
                class="slider animated"
                style="transform: translate3d(0,0%,0); transition-duration: 350ms;"
              >
                <li
                  class="slide selected"
                >
                  <div />
                </li>
              </ul>
            </div>
            <button
              class="control-arrow control-next control-disabled"
              type="button"
            />
          </div>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="searchContainer"
    >
      <div
        class="searchBarContainer"
      >
        <input
          class="searchBar"
          data-testid="searchBar"
          id="searchBar"
          placeholder="Search by name, type, or number!"
          type="text"
          value="A"
        />
      </div>
      <div
        class="carousel"
      >
        <div
          class="carousel carousel-slider"
          style="width: 100%;"
        >
          <button
            class="control-arrow control-prev control-disabled"
            type="button"
          />
          <div
            class="slider-wrapper axis-vertical"
          >
            <ul
              class="slider animated"
              style="transform: translate3d(0,0%,0); transition-duration: 350ms;"
            >
              <li
                class="slide selected"
              >
                <div />
              </li>
            </ul>
          </div>
          <button
            class="control-arrow control-next control-disabled"
            type="button"
          />
        </div>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="searchContainer"
      >
        <div
          class="searchBarContainer"
        >
          <input
            class="searchBar"
            data-testid="searchBar"
            id="searchBar"
            placeholder="Search by name, type, or number!"
            type="text"
            value=""
          />
        </div>
        <div
          class="carousel"
        >
          <div
            class="carousel carousel-slider"
            style="width: 100%;"
          >
            <button
              class="control-arrow control-prev control-disabled"
              type="button"
            />
            <div
              class="slider-wrapper axis-vertical"
            >
              <ul
                class="slider animated"
                style="transform: translate3d(0,0%,0); transition-duration: 350ms;"
              >
                <li
                  class="slide selected"
                >
                  <div />
                </li>
              </ul>
            </div>
            <button
              class="control-arrow control-next control-disabled"
              type="button"
            />
          </div>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="searchContainer"
    >
      <div
        class="searchBarContainer"
      >
        <input
          class="searchBar"
          data-testid="searchBar"
          id="searchBar"
          placeholder="Search by name, type, or number!"
          type="text"
          value=""
        />
      </div>
      <div
        class="carousel"
      >
        <div
          class="carousel carousel-slider"
          style="width: 100%;"
        >
          <button
            class="control-arrow control-prev control-disabled"
            type="button"
          />
          <div
            class="slider-wrapper axis-vertical"
          >
            <ul
              class="slider animated"
              style="transform: translate3d(0,0%,0); transition-duration: 350ms;"
            >
              <li
                class="slide selected"
              >
                <div />
              </li>
            </ul>
          </div>
          <button
            class="control-arrow control-next control-disabled"
            type="button"
          />
        </div>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should start with empty search field 1`] = `
<div
  className="searchContainer"
>
  <div
    className="searchBarContainer"
  >
    <input
      className="searchBar"
      data-testid="searchBar"
      id="searchBar"
      onChange={[Function]}
      placeholder="Search by name, type, or number!"
      type="text"
      value=""
    />
  </div>
  <div
    className="carousel"
  >
    <div
      className="carousel carousel-slider"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <button
        className="control-arrow control-prev control-disabled"
        onClick={[Function]}
        type="button"
      />
      <div
        className="slider-wrapper axis-vertical"
        style={
          Object {
            "height": undefined,
          }
        }
      >
        <ul
          className="slider animated"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "MozTransform": "translate3d(0,0%,0)",
              "MozTransitionDuration": "350ms",
              "MsTransform": "translate3d(0,0%,0)",
              "MsTransitionDuration": "350ms",
              "OTransform": "translate3d(0,0%,0)",
              "OTransitionDuration": "350ms",
              "WebkitTransform": "translate3d(0,0%,0)",
              "WebkitTransitionDuration": "350ms",
              "height": undefined,
              "msTransform": "translate3d(0,0%,0)",
              "msTransitionDuration": "350ms",
              "transform": "translate3d(0,0%,0)",
              "transitionDuration": "350ms",
            }
          }
        >
          <li
            className="slide selected"
            onClick={[Function]}
          >
            <div />
          </li>
        </ul>
      </div>
      <button
        className="control-arrow control-next control-disabled"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
  <div />
</div>
`;
